scripts/minikube.sh

```diff
@@ -78,8 +78,15 @@ fi
 K8S_FEATURE_GATES=${K8S_FEATURE_GATES:-"BlockVolume=true,CSIBlockVolume=true,VolumeSnapshotDataSource=true,ExpandCSIVolumes=true"}
 
 #extra-config for kube https://minikube.sigs.k8s.io/docs/reference/configuration/kubernetes/
-EXTRA_CONFIG=${EXTRA_CONFIG:-"--extra-config=apiserver.enable-admission-plugins=PodSecurityPolicy \
-  --extra-config=kubelet.resolv-conf=/run/systemd/resolve/resolv.conf"}
+EXTRA_CONFIG=${EXTRA_CONFIG:-"--extra-config=apiserver.enable-admission-plugins=PodSecurityPolicy"}
+
+# kubelet.resolv-conf needs to point to a file, not a symlink
+# the default minikube VM has /etc/resolv.conf -> /run/systemd/resolve/resolv.conf
+RESOLV_CONF='/run/systemd/resolve/resolv.conf'
+if [[ "${VM_DRIVER}" == "none" ]]; then
+	RESOLV_CONF="$(readlink -f /etc/resolv.conf)"
+fi
+EXTRA_CONFIG="${EXTRA_CONFIG} --extra-config=kubelet.resolv-conf=${RESOLV_CONF}"
```

Comment:
if the EXTRA_CONFIG which is user-defined already has this param what happens now?

Reply:
I am not sure which would get used by kubeadm. Do you only want to add `--extra-config=kubelet.resolv-conf` in case `EXTRA_CONFIG` is completely empty?

Reply:
we can add it to EXTRA_CONFIG if this flag is not present in it?

Reply:
We can, but it will get more complicated than needed. Who would pass a different `-extra-config=kubelet.resolv-conf=..` parameter?!

Reply:
no idea, but we can pass this as per the code, in that case, this will break isn't it? if we want to fix it later, put a TODO.

Reply:
I'll add a `TODO` note. In case someone reports a problem, we'll fix it at that time.